framework/src/vectorpostprocessors/VolumeHistogram.C

```diff
@@ -0,0 +1,93 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "VolumeHistogram.h"
+
+// libmesh includes
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<VolumeHistogram>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+  params.addParam<unsigned int>("bin_number", 50, "Number of histogram bins");
+  params.addCoupledVar("variable", "Variable to bin the volume of");
+  params.addRequiredParam<Real>("min_value", "Minimum variable value");
+  params.addRequiredParam<Real>("max_value", "Maximum variable value");
+  return params;
+}
+
+VolumeHistogram::VolumeHistogram(const InputParameters & parameters) :
+    ElementVectorPostprocessor(parameters),
+    _nbins(getParam<unsigned int>("bin_number")),
+    _min_value(getParam<Real>("min_value")),
+    _max_value(getParam<Real>("max_value")),
+    _deltaV((_max_value - _min_value) / _nbins),
+    _value(coupledValue("variable")),
+    _bin_center(declareVector(getVar("variable", 0)->name())),
+    _volume(declareVector("n"))
+{
+  if (coupledComponents("variable") != 1)
+    mooseError("VolumeHistogram works on exactly one coupled variable");
+
+  // initialize the bin center value vector
+  _bin_center.resize(_nbins);
+  for (unsigned i = 0; i < _nbins; ++i)
+    _bin_center[i] = (i + 0.5) * _deltaV + _min_value;
+}
+
+void
+VolumeHistogram::initialize()
+{
+  // reset the histogram
+  _volume.assign(_nbins, 0.0);
+}
+
+void
+VolumeHistogram::execute()
+{
+  // loop over quadrature points
+  for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
+  {
+    // compute target bin
+    int bin = (_value[_qp] - _min_value) / _deltaV;
+
+    // add the volume contributed by the current quadrature point
+    if (bin >= 0 && bin < _nbins)
+      _volume[bin] += computeVolume();
+  }
+}
+
+void
+VolumeHistogram::finalize()
+{
+  gatherSum(_volume);
```

Comment:
To be clear, I'm not asking you to change this on this PR.

To answer your question:
`_communicator.sum()` 
I don't see any purpose to wrapping a very, very small subset of the methods available on the communicator. It doesn't make anything "easier" or clearer. I'd rather see that the communicator object is involved and just go look at the available methods. Also if you need to do a sum and then some other operation that's not wrapped in the same post processor should you mix these styles?